qcodes/instrument_drivers/Keysight/Keysight_N5230A.py

```diff
@@ -0,0 +1,160 @@
+
+        self.add_parameter('s_parameter',
+                           label='S parameter',
+                           docstring='Set the measured S parameter, allowed values are "S11", "S12", "S21", and "S22"',
+                           get_cmd=':CALC:PAR:CAT?',
+                           set_cmd=':CALC:PAR:MOD  {}',
+                           get_parser=lambda x: x[1:-1].split(',')[1])
+
+        self.add_parameter('num_points',
+                           label='Number of points',
+                           get_cmd=':SENS:SWE:POIN?',
+                           set_cmd=':SENS:SWE:POIN  {}',
+                           get_parser=int)
+
+        self.add_parameter('num_averages',
+                           label='Number of Averages of a sweep',
+                           get_cmd=':SENS1:AVER:COUN?',
+                           set_cmd=':SENS1:AVER:COUN {}',
+                           get_parser=int)
+
+        self.add_parameter('averaging',
+                           label='Enable Averaging',
+                           get_cmd=':SENS:AVER?',
+                           set_cmd=':SENS:AVER {}',
+                           val_mapping={'on': 1, 'off': 0})
+
+        self.add_parameter('start_sweep',
+                           label='Start Sweep',
+                           get_cmd='*STB?',
+                           set_cmd=(':SENS:SWE:TIME:AUTO ON;'
+                                    ':SENS:AVER:CLE;'
+                                    '*CLS;*ESE 1;'
+                                    ':SENS:SWE:GRO:COUN {};'
+                                    ':SENS:SWE:MODE GRO;'
+                                    '*OPC'),
+                           get_parser=lambda x: int(x) & 32 == 32
+                           )
+
+        self.connect_message()
+
+    def get_idn(self):
+        IDN = self.ask_raw('*IDN?')
+        vendor, model, serial, firmware = map(str.strip, IDN.split(','))
+        IDN = {'vendor': vendor, 'model': model,
+               'serial': serial, 'firmware': firmware}
+        return IDN
+
+    def _get_trace(self, verbose=False):
+
+        self.start_sweep(self.num_averages())
+        while not self.start_sweep():
+            time.sleep(0.5)
+        data = self._get_trace_nowait()
+        self.write('SENS:SWE:MODE CONT')
+
+        return data
+
+    def _get_trace_nowait(self):
+        """This gets the current trace on the VNA without waiting for a sweep to finish"""
+        # Get the current viewing format
+        current_format = self.ask('CALC:FORM?')
+
+        # Change format to smith, and return the current full trace
+        data = self.visa_handle.query_binary_values('CALC:FORM SMIT;:FORM:DATA REAL,64;:CALC1:DATA? FDATA',
+                                                    container=np.array,
+                                                    datatype='d',
+                                                    is_big_endian=True)
+        # Format it correctly
```

Comment:
instead of just saying "formatting it correctly", please, write in the comment explicitly that the returned data array actually contains alternating real and imaginary parts of the values, hence the need for "formatting it correctly"